### PR TITLE
splicectl: Release - Debian/Ubuntu (apt-get/deb) Package build during github action

### PR DESCRIPTION
**Pull request by RileyMcCuen**
## Description

Added a github action that deploys a new apt package with the updated splicectl artifact every time a release is submitted to gh.

## Motivation and Context

This makes it much easier to install and manage splicectl on Debian/Ubuntu machines now, and any other distros that use apt packages.

See [DBAAS-5850](https://splicemachine.atlassian.net/browse/DBAAS-5850?atlOrigin=eyJpIjoiNmQ5ZTNmNWUwMWU4NDYyNTlkZTkzZjY1NzY5ZmI4ODAiLCJwIjoiaiJ9)

## How Has This Been Tested?

I made a fork of this repository and tested the whole process with a manual trigger of the workflow, all of the packages made it to their destination. There is also a test built into the action, where it will use apt-get to install the package and make sure the version that is output by the `splicectl version` command matches the newest release, in effect verifying the new artifact has been installed.

## Checklist

If the pull request includes user-facing changes, extra documentation is required:

- [ ] If the change is user facing, please ensure you add info in one of the [Changelog Inclusions](#changelog-inclusions) sections.

## Changelog Inclusions

- base feature note
  - **BREAKING** note on base feature
  - Basically whatever formatting we have here, just plain-text
	%> command example
- next feature
  - note on next feature

### Additions

Can now install splicectl on Debian and Ubuntu machines using the instructions in the top level readme.

### Breaking Changes

### CODE REVIEW

**Review by splicemaahs — APPROVED** [review]
We need to remember to put the GPG keys in place before the next release.

File: .github/workflows/apt-release.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Build Package
+        env:
+          KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+        run: |
+          sudo gem install deb-s3
+          # Set release version based on github ref
+          # RELEASE_VERSION=v0.1.2
```
Comment (skolobov) [inline .github/workflows/apt-release.yaml:24]: Shouldn't we delete this line?

File: .github/workflows/apt-release.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    needs: release
+    steps:
+      - 
+        name: Checkout
+        uses: actions/checkout@v2
+      - 
+        name: Test package
+        run: |
+          # Set release version based on github ref
+          RELEASE_VERSION=v0.1.2
```
Comment (skolobov) [inline .github/workflows/apt-release.yaml]: Is it hard-coded here?

**Review by skolobov — APPROVED** [review]

File: .github/workflows/apt-release.yaml (hunk 2 above)
Comment (RileyMcCuen) [reply .github/workflows/apt-release.yaml]: Yes, good catch. Fixed.

File: .github/workflows/apt-release.yaml (hunk 1 above)
Comment (RileyMcCuen) [reply .github/workflows/apt-release.yaml:24]: I left it in for future reference of what the expected format should be. The bash script to get the tag below that line is a bit of an eyeful.

**Comment by RileyMcCuen** [comment]
> We need to remember to put the GPG keys in place before the next release.

I have added the keys to s3, gh actions, and vault as well.
